systemtests/src/test/java/io/enmasse/systemtest/sharedinfra/SharedInfraDefaultsTest.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.enmasse.systemtest.sharedinfra;
+
+import io.enmasse.api.model.MessagingInfra;
+import io.enmasse.api.model.MessagingInfraCondition;
+import io.enmasse.api.model.MessagingTenant;
+import io.enmasse.api.model.MessagingTenantCondition;
+import io.enmasse.systemtest.TestTag;
+import io.enmasse.systemtest.annotations.DefaultMessagingInfra;
+import io.enmasse.systemtest.annotations.DefaultMessagingTenant;
+import io.enmasse.systemtest.bases.TestBase;
+import io.enmasse.systemtest.bases.isolated.ITestIsolatedSharedInfra;
+import io.enmasse.systemtest.messaginginfra.resources.MessagingInfraResourceType;
+import io.enmasse.systemtest.messaginginfra.resources.MessagingTenantResourceType;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+@Tag(TestTag.ISOLATED_SHARED_INFRA)
+public class SharedInfraDefaultsTest extends TestBase implements ITestIsolatedSharedInfra {
+
+    @Test
+    @DefaultMessagingInfra
+    public void testDefaultInfra() {
+        MessagingInfra infra = infraResourceManager.getDefaultInfra();
```

Comment:
For this to be more useful, would it be an idea to be able to specify a signature like this:

```
public void testDefaultInfra(MessagingInfra default) {
}
```

And to have multiple defaults:

```
public void testFoo(MessagingInfra infra1, MessagingTenant tenant1) {}
```
?

Reply:
@lulf junit test methods are not like a normal methods -> it does not have method signatures like standard methods, you can only pass what Extension context allows you. And if you need something more you want to use parametrized test, but it is not easy for use for all, My idea was provide the most basic stuff. I like the idea that I can write easy test without taking care about default resources.

Reply:
Ok, so maybe not in the method signature... I was wondering if we could maybe use reflection + annotations to support the following:

```
public class MyTest {
   @ Default
   public MessagingInfra infra

   @ Test  
    public void mytest() {
      // infra is automatically created.
    }
}
```

Reply:
so what about this solution? `@DefaultMessagingInfra` if you want just base and `DefaultMessagingInfra(infra = instance)` if you want your own specified.

Reply:
but Im not sure it is posible in annotation :) let me investigate it

Reply:
Sure :) Do you want to merge this now, or should we take the time to explore some options before we go all in on a specific approach?